classes/PaymentModule.php

```diff
@@ -1060,6 +1060,10 @@ protected function createOrderFromCart(
             throw new PrestaShopException('Can\'t save Order');
         }
 
+        // update cart's order id
+        $cart->id_order = $order->id;
```

Comment:
If we want to be safer and not only rely on this module and expect every Order creation to be made here, we should do this in `Order::add` and/or `Order::update` so that any modif saved via the ObjectModel ensure that the cart is always correctly updated (even if it's not created via a PaymentModule)